### ISSUE TITLE
[Snowflake] Unable to use CTEs in `COPY INTO <location>`

### ISSUE THREAD

**Issue by mroy-seedbox** [sqlfluff/sqlfluff#6310]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

This works:
```sql
COPY INTO @my_stage/path/to/file.json.gz
FROM
(
  SELECT 123
);
```

This triggers a parsing error:
```sql
COPY INTO @my_stage/path/to/file.json.gz
FROM
(
  WITH my_cte AS (
    SELECT 1
  )
  SELECT 123
);
```

`Line 2, Position 1: Found unparsable section: 'FROM/n(/n  WITH my_cte AS (/n    SELECT 1/n ...'sqlfluff(PRS)`

### Expected Behaviour

No parsing error occurs.

The inside of the `FROM (...)` should probably be parsed as a completely separate/individual/standalone query.

### Observed Behaviour

See above.

### How to reproduce

See above.

### Dialect

Snowflake

### Version

```
sqlfluff, version 3.2.2
Python 3.12.6
```

### Configuration

Not configuration related.

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
